mern-ecommerce/controllers/auth.js

```diff
@@ -62,20 +62,20 @@ exports.requireSignin = expressJwt({
   userProperty: 'auth',
 });
 
-exports.isAuth = (req, res, next) => {
-  let user = req.profile && req.auth && req.profile._id == req.auth._id;
-  if (!user) {
+exports.isAdmin = (req, res, next) => {
+  if (req.profile.role === 0) {
     return res.status(403).json({
-      error: 'Access denied',
+      error: 'Admin resource! Access denied',
     });
   }
   next();
 };
 
-exports.isAdmin = (req, res, next) => {
-  if (req.profile.role === 0) {
+exports.isAuth = (req, res, next) => {
+  let user = req.profile && req.auth && req.profile._id == req.auth._id;
+  if (!user) {
```

Comment:
The condition `if (!user)` is not checking if the user is authenticated or not. It's only checking if the `user` variable is truthy or falsy. It would be better to check if the user is authenticated by verifying the existence of a valid token or session. Additionally, this condition is present in multiple places in the code, it would be better to create a utility function to handle this check.